src/filter_tree/ft_to_rsq.c

```diff
@@ -107,6 +108,40 @@ static RSQNode *_FilterTreeToInQueryNode
 	ASSERT(attribute == true);
 
 	SIValue list = inOp->op.children[1]->operand.constant;
+	if(AR_EXP_IsOperation(inOp->op.children[1]) && strcmp(inOp->op.children[1]->op.f->name, "range") == 0) {
+		AR_ExpNode *range = inOp->op.children[1];
+		int64_t step     = 1;
+		// extracting information from range function
+		// extracting: range beginning, range end and optional range step
+		SIValue v = AR_EXP_Evaluate(range->op.children[0], NULL);
+		if(SI_TYPE(v) != T_INT64) {
+			Error_SITypeMismatch(v, T_INT64);
+			return RediSearch_CreateEmptyNode(idx);
+		}
+		int64_t from = v.longval;
+		v = AR_EXP_Evaluate(range->op.children[1], NULL);
+		if(SI_TYPE(v) != T_INT64) {
+			Error_SITypeMismatch(v, T_INT64);
+			return RediSearch_CreateEmptyNode(idx);
+		}
+		int64_t to = v.longval;
+		if(range->op.child_count == 3) {
+			v = AR_EXP_Evaluate(range->op.children[2], NULL);
+			if(SI_TYPE(v) != T_INT64) {
+				Error_SITypeMismatch(v, T_INT64);
+				return RediSearch_CreateEmptyNode(idx);
+			}
+			step = v.longval;
+		}
+		to      += step;
+		int64_t current  = (to >= from && step > 0) || (to <= from && step < 0) ? from : to;
+		RSQNode *U = RediSearch_CreateUnionNode(idx);
+		while(current != to) {
+			RediSearch_QueryNodeAddChild(U, RediSearch_CreateNumericNode(idx, field, current, current, true, true));
+			current += step;
+		}
+		return U;
```

Comment:
This is too much,
I think we should in this case detect the fact that a `range` function is used and replace it with an internal function named `iterate` which will return an `SIValue` of type `iterator` I believe this can be used in other places as well.